hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -200,7 +208,7 @@ return_type GenericSystem::read()
   for (size_t i = 0; i < joint_states_.size(); ++i) {
     for (size_t j = 0; j < joint_states_[i].size(); ++j) {
       if (!std::isnan(joint_commands_[i][j])) {
-        joint_states_[i][j] = joint_commands_[i][j];
+        joint_states_[i][j] = joint_commands_[i][j] + state_following_offset_;
```

Comment:
Should we only apply the offset to the position interface .? otherwise we may always have values > 0.0 for vel/acc

Reply:
Oh, of course! You are right! I made a mistake when resolving conflict.

Reply:
I just updated the code and also adjusted tests to detect this. Thanks once again.